src/views/Player.vue

```diff
@@ -103,11 +137,123 @@ export default defineComponent({
+          },
+          value: state.numCorrect,
+        },
+        {
+          name: "Wrong",
+          icon: {
+            source: "wrong",
+            class:
+              "text-red-500 h-8 bp-360:h-8 bp-500:h-10 lg:h-11 w-6 bp-360:w-6 bp-500:w-6 md:w-7 lg:w-8 mx-1 place-self-center",
+          },
+          value: state.numWrong,
+        },
+      ];
+    });
+
+    /**
+     * progress value (0-100) to be passed to the Scorecard component
+     */
+    const scorecardProgress = computed(() => {
+      const totalAttempted = state.numCorrect + state.numWrong;
+      if (totalAttempted == 0) return null;
+      return (state.numCorrect / totalAttempted) * 100;
+    });
+
+    /**
+     * number of questions that have been answered
+     */
+    const numQuestionsAnswered = computed(() => {
+      return state.numCorrect + state.numWrong;
+    });
+
+    const numSurveyQuestions = computed(() => {
+      let count = 0;
+      state.questions.forEach((itemDetail) => {
+        if (itemDetail.survey) count += 1;
+      });
+      return count;
+    });
+
+    const areAllQuestionsSurvey = computed(() => {
+      return numSurveyQuestions.value == state.questions.length;
+    });
+
+    function calculateScorecardMetrics() {
+      let index = 0;
+      state.questions.forEach((itemDetail) => {
+        updateNumCorrectWrongSkipped(itemDetail, state.responses[index].answer);
+        index += 1;
+      });
+    }
+
+    function updateNumCorrectWrongSkipped(itemDetail: any, userAnswer: any) {
+      if (itemDetail.survey) {
+        return;
+      }
+      if (
+        itemDetail.type == "mcq" &&
+        userAnswer != null &&
+        userAnswer.length > 0
+      ) {
+        const correctAnswer = itemDetail.correct_answer;
+        isEqual(userAnswer, correctAnswer)
+          ? (state.numCorrect += 1)
+          : (state.numWrong += 1);
+      } else if (
+        itemDetail.type == "checkbox" &&
+        userAnswer != null &&
+        userAnswer.length > 0
+      ) {
+        // for checkbox questions, check if the answers match exactly
+        const correctAnswer = itemDetail.correct_answer;
+
+        isEqual(userAnswer, correctAnswer)
+          ? (state.numCorrect += 1)
+          : (state.numWrong += 1);
+      } else if (
+        itemDetail.type == "subjective" &&
+        userAnswer != null &&
+        userAnswer.trim() != ""
+      ) {
+        // for subjective questions, as long as the viewer has given any answer
+        // their response is considered correct
+        state.numCorrect += 1;
+      }
+    }
+
+    /**
+     * remove the scorecard, display last question and remove the confetti
+     */
+    function restartQuiz() {
+      state.isScorecardShown = false;
+      state.currentQuestionIndex -= 1;
+      state.numCorrect = 0;
+      state.numWrong = 0;
+      resetConfetti();
+    }
```

Comment:
not restarting as I mentioned before - just moving them back to the last answered question